### ISSUE TITLE
Reversals for legacy points logs don't work for some entities

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/hooks-api#142]
While working on #99 and #80, I realized that the back-compat code for reversals in the points reactor is broken. When querying the points logs, we don't take into account the fact that some of the entity slugs that we are using today don't match up with the slugs that we were using before. Specifically, the comment and post entity slugs are now namespaced like `post\post-type`, whereas before they would have just been like `post`.

This is really a bigger issue than just this one case, it will affect anything that takes the points logs into account, like the retroactive API (#1).

**Comment by JDGrimes** [comment]
I was thinking that maybe we should have a registered list of entity entity slugs that need to be mapped for legacy purposes. But now I'm wondering whether that is really necessary. We could just parse namespaced entity slugs and use just the namespace, since that will work for both the post and comment entities. That wouldn't provide a simple way for any other legacy mapping that anyone else might need though. Then again, I don't think that is a huge concern at the moment.

**Comment by JDGrimes** [comment]
The problem here is that we'd need to either run two queries or use an `OR` sub-condition in our meta query. However, I'm not sure that we can do `OR` sub-conditions until WordPress 4.3, and currently WordPoints supports 4.2. It turns out that [this was actually introduced in 4.1](https://core.trac.wordpress.org/ticket/29642), so we are good there.

**Comment by JDGrimes** [comment]
I suppose that we could even limit this to only particular events that we run the back-compat query for, since it isn't needed for all of them. But I guess that would need an event map or something. Anyway, that would be for another ticket.

**Comment by JDGrimes** [comment]
Another thing is that currently we are querying by all of the entities that an event supports. But we should really only be querying based on the primary entity, since that is the only one that will have been logged by the old points hooks. When the stateful args are included, this will prevent our query from ever turning up any legacy logs to reverse.

I think this should also change for querying the points logs from the new points reactions as well, but maybe this should really be for a separate ticket.

**Comment by JDGrimes** [comment]
Also, within the legacy code we ought to be marking these legacy points logs as auto-reversed after #80, because otherwise they will continue to come up as unreversed even after they have been.

**Comment by JDGrimes** [comment]
I just realized that this affects the event slugs two, since some of them are also namespaced. It isn't as simple as just ripping the post type off of the back though, because I think we've changed the base slugs for some of the events. So we'll need to have some sort of even map after all.

**Comment by JDGrimes** [comment]
| Old Slug | New Slug |
| --- | --- |
| `register` | `user_register` |
| `post_publish` | `post_publish\*`, `media_upload` |
| `comment_approve` | `comment_leave\*` |
| `comment_received` | `comment_leave\*` |
| `periodic` | `user_visit` |

As far as the `media_upload` event goes, see #45. Post publish is the only one of these that is going to possibly map correctly on its own. The others will need explicit mapping.

**Comment by JDGrimes** [comment]
I decided that the best thing to do was just to let each reactor imported from the legacy hooks specify the legacy log type slug. We'll have to keep that in mind in #23.
